### DISCUSSION TITLE
Telegram notification

### DISCUSSION THREAD

**Discussion by andreamazzoni** [garethgeorge/backrest#460] [Show and tell]
If anyone is interested, this is the script for sending notifications via telegram, in a chat of your choice, using a bot:

```
chatid="<your Chat ID>"
api="<your bot API token>"
log={{ .ShellEscape .Summary }}
curl -s \
  --data parse_mode=HTML \
  --data chat_id=$chatid \
  --data text="$log" \
  "https://api.telegram.org/bot$api/sendMessage"
```
I found it here: https://elblogdelazaro.org/posts/2024-07-15-backrest-notificaciones-via-telegram/
Tested and working.

If you don't know how to retrieve a chatid the best way is to send a message to your new bot and invoke a GET here: `https://api.telegram.org/bot<your bot API token>/getUpdates`

**Comment by robflate** [comment]
FYI, you can use the built in Shoutrrr HOOK and add this as the URL;

`telegram://<token>@telegram?channels=<chat-id>`

https://containrrr.dev/shoutrrr/v0.8/services/telegram/
